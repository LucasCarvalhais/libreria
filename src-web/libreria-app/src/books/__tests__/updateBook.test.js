import React from 'react';
import { shallow } from 'enzyme';
import UpdateBook from '../UpdateBook';

describe('UpdateBook', () => {
    test('should show the form to search book', () => {
        const component = shallow(<UpdateBook />).setState({
            successSearch: false,
            successUpdate: false,
            errorSearch: null,
            errorUpdate: null,
        });
        expect(component.find('FormSearch')).toHaveLength(1);
    });

    test('should show the form to update book', () => {
        const component = shallow(<UpdateBook />).setState({
            successSearch: true,
            successUpdate: false,
            errorSearch: null,
            errorUpdate: null,
        });
        expect(component.find('FormBook')).toHaveLength(1);
    });

    test('should show the success message', () => {
        const component = shallow(<UpdateBook />).setState({
            successSearch: true,
            successUpdate: true,
            errorSearch: null,
            errorUpdate: null,
        });
        expect(component.find('.message')).toHaveLength(1);
    });

    test('should show the error message', () => {
        const component = shallow(<UpdateBook />).setState({
            successSearch: false,
            successUpdate: false,
            errorSearch: 'Error',
            errorUpdate: null,
        });
        expect(component.find('ErrorMessage')).toHaveLength(1);
    });
});
