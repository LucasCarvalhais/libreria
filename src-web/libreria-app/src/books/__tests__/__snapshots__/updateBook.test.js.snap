// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`updateBook has a valid snapshot 1`] = `
<div>
  <h1>
    Atualizar el libro
  </h1>
  <form
    className="formulario"
    onSubmit={[Function]}
  >
    <label>
      <span
        className="legend"
      >
        Pesquisar libro: 
      </span>
      <input
        className="inputForm"
        name="bookId"
        onChange={[Function]}
        type="number"
      />
    </label>
    <input
      className="submitButton"
      type="submit"
      value="Pesquisar"
    />
  </form>
  <div
    className="message"
  >
    <p>
      Por favor seleciona un c√≥digo del libro y clica en "Pesquisar".
    </p>
  </div>
</div>
`;
